tests/Unit/DataStructures/Tensor/Expressions/Test_DataTypeSupport.cpp

```diff
@@ -115,7 +115,7 @@ SPECTRE_TEST_CASE("Unit.DataStructures.Tensor.Expression.DataTypeSupport",
   test_is_supported_number_datatype<double, true>();
```

Comment:
Generalyquestion for this file: this is a very large number of similar checks. Is it possible to generalize them to use things like `tmpl::for_each`-type loops to do the checks? E.g. you could have a `tmpl::list<tmpl::list<std::complex<double>, std::true_type>, ...>`? Or, there might be an even easier way to generalize it, not sure :) But given the length of this file, it might be worth thing about :)

Reply:
That's a good idea. I will try to make it more concise and general!

Reply:
I was able to condense parts of the file a bit but not a ton